fedimint-server/src/config.rs

```diff
@@ -410,6 +401,19 @@ pub struct PeerServerParams {
 }
 
 impl ServerConfigParams {
+    /// Generates denominations as powers of 2 until a `max`
+    pub fn gen_denominations(max: Amount) -> Vec<Amount> {
+        let mut amounts = vec![];
+
+        let mut denomination = Amount::from_msat(1);
+        while denomination < max {
+            amounts.push(denomination);
+            denomination = denomination * 2;
+        }
```

Comment:
Good use case for [`take_while`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take_while).

```rust
(0..64).map(|power| Amount::from_msat(1u64 << power))
    .take_while(|&amt| amt < max)
    .collect()
```

Reply:
Kind of an unnecessary optimization though and wouldn't work with other powers.